features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/AsyncApiConsole/AsyncApiConsole.jsx

```diff
@@ -0,0 +1,256 @@
+import Progress from '../../../Shared/Progress';
+import AsyncApiUI from './AsyncApiUI';
+
+const useStyles = makeStyles((theme) => ({
+    buttonIcon: {
+        marginRight: 10,
+    },
+    paper: {
+        margin: theme.spacing(1),
+        padding: theme.spacing(1),
+    },
+    grid: {
+        marginTop: theme.spacing(4),
+        marginBottom: theme.spacing(4),
+        paddingRight: theme.spacing(2),
+        justifyContent: 'center',
+    },
+    userNotificationPaper: {
+        padding: theme.spacing(2),
+    },
+    titleSub: {
+        marginLeft: theme.spacing(2),
+        paddingTop: theme.spacing(2),
+        paddingBottom: theme.spacing(2),
+    },
+}));
+
+export default function AsyncApiConsole() {
+    const classes = useStyles();
+    const { api } = useContext(ApiContext);
+    const [securitySchemeType, setSecurityScheme] = useState('OAUTH');
+    const [selectedEnvironment, setSelectedEnvironment] = useState();
+    const [productionAccessToken, setProductionAccessToken] = useState();
+    const [sandboxAccessToken, setSandboxAccessToken] = useState();
+    const [environments, setEnvironments] = useState();
+    const [scopes, setScopes] = useState([]);
+    const [labels, setLabels] = useState();
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+    const [selectedKeyType, setSelectedKey] = useState('PRODUCTION');
+    const [keys, setKeys] = useState([]);
+    const [sandboxApiKey, setSandboxApiKey] = useState('');
+    const [productionApiKey, setProductionApiKey] = useState('');
+    const environmentObject = api.endpointURLs;
+    const [URLs, setURLs] = useState(environmentObject[0].URLs);
+    const [notFound, setNotFound] = useState(false);
+
+    const user = AuthManager.getUser();
+
+    useEffect(() => {
+        const apiID = api.id;
+        const apiClient = new Api();
+        const promiseAPI = apiClient.getAPIById(apiID);
+        promiseAPI
+            .then((apiResponse) => {
+                const apiData = apiResponse.obj;
+                if (apiData.endpointURLs) {
+                    const environment = apiData.endpointURLs.map((endpoint) => {
+                        return endpoint.environmentName;
+                    });
+                    setEnvironments(environment);
+                }
+                if (apiData.labels) {
+                    const Label = apiData.labels.map((label) => {
+                        return label.name;
+                    });
+                    setLabels(Label);
+                }
+                if (apiData.scopes) {
+                    const scopeList = apiData.scopes.map((scope) => {
+                        return scope.name;
+                    });
+                    setScopes(scopeList);
+                }
+            })
+            .catch((error) => {
+                if (process.env.NODE_ENV !== 'production') {
+                    console.error(error);
+                }
```

Comment:
It's better to show an error to the user here as well.
```
Alert.error(error);
```

Reply:
fixed